### PR TITLE
Fix quota consuming

### LINKED ISSUE #20018: Show tables command will take 2 quota count.
(you don't have to strictly follow this form)

**Describe the bug**
Command suchs show tables/databases/processlist etc will take 2 quota count.

**Does it reproduce on recent release?**
version 20.13.3.3

**How to reproduce**
```
1. create quota

CREATE QUOTA q1 KEYED BY user_name FOR INTERVAL 1 minute MAX queries = 5 TO default

2. show current quota usage

select quota_name, end_time, queries, max_queries from system.quotas_usage;

┌─quota_name─┬────────────end_time─┬─queries─┬─max_queries─┐
│ q1         │ 2021-01-19 14:40:00 │       0 │           5 │

3. send show command
show tables;

4. show quota usage
┌─quota_name─┬────────────end_time─┬─queries─┬─max_queries─┐
│ q1         │ 2021-01-19 14:40:00 │       2 │           5 │
```

**Expected behavior**

* show tables/databases/processlist take 1 quota count
* show privileges take 0 quota count 

**Analysis**

Sqls such as show tables/databases/processlist/privileges will be interpreted to select sql, so will take another quota count.

### PR DESCRIPTION
**Pull request by vitlibar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Improvement

Changelog entry:
1. SHOW TABLES is now considered as one query in the quota calculations, not two queries.
2. SYSTEM queries now consume quota.
3. Fix calculation of interval's end in quota consumption.

This PR fixes https://github.com/ClickHouse/ClickHouse/issues/20018

### CODE REVIEW

File: src/Access/EnabledQuota.cpp [hunk 1]
```diff
@@ -52,22 +48,24 @@ struct EnabledQuota::Impl
             return end;
         }
 
+        bool initial_end = !end_loaded.count(); /// whether the end of the interval wasn't calculated before
         const auto duration = interval.duration;
 
         do
         {
             end = end + (current_time - end + duration) / duration * duration;
```
Comment (nikitamikhaylov) [inline src/Access/EnabledQuota.cpp:56]: I don't understand what do you want to do in this line. `/` and `*` have the same priority.

File: src/Access/EnabledQuota.cpp [hunk 2]
```diff
@@ -52,22 +48,24 @@ struct EnabledQuota::Impl
...
             end = std::chrono::system_clock::time_point{end_loaded};
         }
         while (current_time >= end);
 
-        if (counters_were_reset)
-            *counters_were_reset = true;
+        if (!initial_end)
+        {
+            boost::range::fill(interval.used, 0);
+            if (counters_were_reset)
```
Comment (nikitamikhaylov) [inline src/Access/EnabledQuota.cpp:66]: Could you please add some comments about `counters_were_reset` variable ?

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
I read the whole file and found a piece of useless code. Sorry, if I am wrong.

**Comment by vitlibar** [comment]
> I read the whole file and found a piece of useless code. Sorry, if I am wrong.

Yes, it seems to be useless, thank you.

File: src/Access/EnabledQuota.cpp (hunk 1 above)
Comment (vitlibar) [reply src/Access/EnabledQuota.cpp:56]: '/' and then '*' is a way to remove the remainder. I think I'll add a comment here.

File: src/Access/EnabledQuota.cpp (hunk 2 above)
Comment (vitlibar) [reply src/Access/EnabledQuota.cpp]: done

**Comment by vitlibar** [comment]
There was a problem unrelated to this PR:
https://clickhouse-test-reports.s3.yandex.net/20106/5f8a6ab9c109a82ab044b6ee573f86320175839a/integration_tests_(asan).html#fail1

```
Failed to start cluster: 
Command '['docker-compose', '--project-name', 'roottestgrpcprotocolssl', '--file', '/compose/docker_compose_net.yml', '--file', '/ClickHouse/tests/integration/test_grpc_protocol_ssl/_instances/node/docker-compose.yml', 'up', '-d', '--no-recreate']' returned non-zero exit status 1.
```

**Pull request merged**